scripts/caravan/caravan.lua

```diff
@@ -164,40 +179,64 @@ py.on_event(py.events.on_entity_clicked(), function(event)
         position = event.cursor_position,
         limit = 1,
         collision_mask = {object = true, player = true, train = true, resource = true, floor = true, transport_belt = true, ghost = true}
     }[1]
-    if entity then
-        if entity.operable then storage.make_operable_next_tick[#storage.make_operable_next_tick + 1] = entity end
-        entity.operable = false
-        if only_outpost and entity.name ~= prototype.outpost then return end
-        if entity == caravan_data.entity or entity.surface ~= caravan_data.entity.surface then return end
-        schedule[#schedule + 1] = {
-            localised_name = {"caravan-gui.entity-position", entity.prototype.localised_name, math.floor(entity.position.x), math.floor(entity.position.y)},
-            entity = entity,
-            position = entity.position,
-            actions = {}
-        }
-    elseif not only_outpost then
-        local position = event.cursor_position
-        schedule[#schedule + 1] = {
-            localised_name = {"caravan-gui.map-position", math.floor(position.x), math.floor(position.y)},
-            position = position,
-            actions = {}
-        }
+    if storage.last_opened_action[player.index] then
+        if entity then
+            if entity.operable then storage.make_operable_next_tick[#storage.make_operable_next_tick + 1] = entity end
+            entity.operable = false -- Prevents the player from opening the gui of the clicked entity
+            if (entity.name == "outpost") or (entity.name == "outpost-aerial") then
+                local action_id = storage.last_opened_action[player.index].action_id
+                interrupt_data.conditions[action_id].entity = entity
+                interrupt_data.conditions[action_id].localised_name = {"caravan-actions.at-outpost2", {"caravan-gui.entity-position", entity.localised_name, entity.position.x, entity.position.y}}
+            end
+        end
     else
-        return
+        if entity then
+            if entity.operable then storage.make_operable_next_tick[#storage.make_operable_next_tick + 1] = entity end
+            entity.operable = false -- Prevents the player from opening the gui of the clicked entity
+            if only_outpost and entity.name ~= prototype.outpost then return end
+            if caravan_data and (entity == caravan_data.entity or entity.surface ~= caravan_data.entity.surface) then return end
+            schedule[#schedule + 1] = {
+                localised_name = {"caravan-gui.entity-position", entity.prototype.localised_name, math.floor(entity.position.x), math.floor(entity.position.y)},
+                entity = entity,
+                position = entity.position,
+                actions = {}
+            }
+        elseif not only_outpost then
+            local position = event.cursor_position
+            schedule[#schedule + 1] = {
+                localised_name = {"caravan-gui.map-position", math.floor(position.x), math.floor(position.y)},
```

Comment:
This should be `entity.gps_tag`
Using gps_tag allows this code to display the caravan's current planet, which is necessary for pySEx.